perun-rpc/src/main/java/cz/metacentrum/perun/rpc/methods/MembersManagerMethod.java

```diff
@@ -141,19 +142,43 @@ public Member call(ApiCaller ac, Deserializer parms) throws PerunException {
 	 * @param sponsor int sponsor's ID
 	 * @return RichMember newly created sponsored member
 	 */
+	/*#
+	 * Creates a new sponsored member in a given VO and namespace.
+	 *
+	 * Can be called either by a user with role SPONSOR, in that case the user becomes the sponsor,
+	 * or by a user with role REGISTRAR that must specify the sponsoring user using ID.
+	 *
+	 * @param firstName first name - mandatory
+	 * @param lastName last name - mandatory
+	 * @param titleBefore titles before the name - optionally
+	 * @param titleAfter titles after the name - optionally
+	 * @param password String password
+	 * @param vo int VO ID
+	 * @param namespace String namespace selecting remote system for storing the password
+	 * @param sponsor int sponsor's ID
+	 * @return RichMember newly created sponsored member
+	 */
 	createSponsoredMember {
 		@Override
 		public RichMember call(ApiCaller ac, Deserializer params) throws PerunException {
 			ac.stateChangingCheck();
-			String guestName = params.readString("guestName");
 			String password = params.readString("password");
 			Vo vo =  ac.getVoById(params.readInt("vo"));
 			String namespace = params.readString("namespace");
 			User sponsor = null;
 			if(params.contains("sponsor")) {
 				sponsor = ac.getUserById(params.readInt("sponsor"));
 			}
-			return ac.getMembersManager().createSponsoredMember(ac.getSession(), vo, namespace, guestName, password, sponsor);
+			Map<String, String> name = new HashMap<>();
+			if (params.contains("firstName")) {
```

Comment:
I would prefer to differentiate behavior based on presence of ``guestName`` rather than ``firstName``. It is used like this later in BL logic and conforms original implementation.
Like this you could pass ``guestName`` and ``lastName`` and it would work and just ignored ``lastName`` value without letting you know. You can probably create third branch where you throw exception if wrong combination of params is passed. Like:
```java
if (params.contains("guestName")) {
...
} else if (params.contains("firstName") && params.contains("lastName")) {
...
} else {
throw new Exception()
}